### PR TITLE
[nrf fromlist] drivers: nrf: rram: Support TF-M

### PR DESCRIPTION
**Pull request by SebastianBoe**
fromlist: https://github.com/zephyrproject-rtos/zephyr/pull/71825

Non-secure images cannot reference NRF_RRAMC_NS because NRF_RRAMC_NS
does not exist.

TF-M will configure RRAMC according to these Kconfig's before booting
the non-secure image so we ifdef out this code.

Also, rewrite the implementation of commit_changes to also work when
the commit task is not available.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: drivers/flash/soc_flash_nrf_rram.c [hunk 1]
```diff
@@ -58,6 +58,30 @@ BUILD_ASSERT((PAGE_SIZE % (WRITE_LINE_SIZE) == 0),
...
+ *   .write_buff_size = WRITE_BUFFER_SIZE
+ * };
+ *
+ * nrf_rramc_ready_next_timeout_t params = {
+ *   .value = CONFIG_NRF_RRAM_READYNEXT_TIMEOUT_VALUE,
+ *   .enable = true,
+ * };
+ *
+ * For more details see NCSDK-26982.
+ */
```
Comment (de-nordic) [inline drivers/flash/soc_flash_nrf_rram.c:79]: No conditional comments, looks strange, like leftover. Just do comment above the `#if !defined()`, should be enough

File: drivers/flash/soc_flash_nrf_rram.c [hunk 2]
```diff
@@ -85,29 +109,54 @@ static inline bool is_within_bounds(off_t addr, size_t len, off_t boundary_start
 }
 
 #if WRITE_BUFFER_ENABLE
-static void commit_changes(size_t len)
+static void commit_changes(off_t addr, size_t len)
 {
+#ifdef NRF_RRAMC
```
Comment (de-nordic) [inline drivers/flash/soc_flash_nrf_rram.c:114]: Where is this one coming from?

File: drivers/flash/soc_flash_nrf_rram.c (hunk 1 above)
Comment (SebastianBoe) [reply drivers/flash/soc_flash_nrf_rram.c:79]: OK

**Review by vili-nordic — APPROVED** [review]

File: drivers/flash/soc_flash_nrf_rram.c [hunk 3]
```diff
@@ -85,29 +105,53 @@ static inline bool is_within_bounds(off_t addr, size_t len, off_t boundary_start
...
 	}
+#endif
 
 	if ((len % (WRITE_BUFFER_MAX_SIZE)) == 0) {
 		/* Our last operation was buffer size-aligned, so we're done. */
 		return;
 	}
 
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
 	nrf_rramc_task_trigger(NRF_RRAMC, NRF_RRAMC_TASK_COMMIT_WRITEBUF);
```
Comment (nvlsianpu) [inline drivers/flash/soc_flash_nrf_rram.c:123]: ARG_UNUSED(addr);

**Review by nvlsianpu — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

File: drivers/flash/soc_flash_nrf_rram.c (hunk 3 above)
Comment (SebastianBoe) [reply drivers/flash/soc_flash_nrf_rram.c:123]: will fix :)

**Review by jaz1-nordic — APPROVED** [review]

**Review by mswarowsky — APPROVED** [review]

**Pull request merged**